### PR TITLE
Add Hover to TabItem

### PR DESCRIPTION
**Pull request by wildan-m**
@Ollyws, @MariaHCD, @garrettmknight 

### Details

Add tab item styling when hovered

### Fixed Issues

$ https://github.com/Expensify/App/issues/26648
PROPOSAL: https://github.com/Expensify/App/issues/26648#issuecomment-1709891162

### Tests

Web / Desktop
1. Click FAB
2. Choose Request Money
3. Select `Scan` tab as the initial tab
4. Hover the mouse to `Manual` tab
5. Verify that `Manual` tab background color is equal to themeColors.highlightBG / darkHighlightBackground / #07271F
6. Verify that the icon and label color are the same as with the active tab (Scan tab)
7. Click `Manual` tab
8. Hover the mouse to `Scan` tab
5. Verify that `Scan` tab background color is equal to themeColors.highlightBG / darkHighlightBackground / #07271F
6. Verify that the icon and label color are the same as with the active tab (Manual tab)

Native / Mobile Web
1. Click FAB
2. Choose Request Money
3. Select `Scan` tab as the initial tab
4. Select `Manual` tab
5. Verify that there is no error

- [x] Verify that no errors appear in the JS console

### Offline tests

Web / Desktop
1. Click FAB
2. Choose Request Money
3. Select `Scan` tab as the initial tab
4. Hover the mouse to `Manual` tab
5. Verify that `Manual` tab background color is equal to themeColors.highlightBG / darkHighlightBackground / #07271F
6. Verify that the icon and label color are the same as with the active tab (Scan tab)
7. Click `Manual` tab
8. Hover the mouse to `Scan` tab
5. Verify that `Scan` tab background color is equal to themeColors.highlightBG / darkHighlightBackground / #07271F
6. Verify that the icon and label color are the same as with the active tab (Manual tab)

Native / Mobile Web
1. Click FAB
2. Choose Request Money
3. Select `Scan` tab as the initial tab
4. Select `Manual` tab
5. Verify that there is no error

### QA Steps

Web / Desktop
1. Click FAB
2. Choose Request Money
3. Select `Scan` tab as the initial tab
4. Hover the mouse to `Manual` tab
5. Verify that `Manual` tab background color is equal to themeColors.highlightBG / darkHighlightBackground / #07271F
6. Verify that the icon and label color are the same as with the active tab (Scan tab)
7. Click `Manual` tab
8. Hover the mouse to `Scan` tab
5. Verify that `Scan` tab background color is equal to themeColors.highlightBG / darkHighlightBackground / #07271F
6. Verify that the icon and label color are the same as with the active tab (Manual tab)

Native / Mobile Web
1. Click FAB
2. Choose Request Money
3. Select `Scan` tab as the initial tab
4. Select `Manual` tab
5. Verify that there is no error

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/11847279/3b82b9ad-4469-4740-bd41-60b7b1568834

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/11847279/b16d740e-b108-4e34-81ad-b174f3e40312

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/11847279/74c594b5-bce5-4c29-9152-4447e54c0bdd

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/11847279/60fe705b-c0fc-4e36-9f87-51e9981d5fc4

</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/assets/11847279/8aeaaa44-0392-49da-a099-a701da043009

</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/11847279/d233d409-2dfb-4fd9-b9a2-51b2bd0336d1

</details>

### CODE REVIEW

File: src/components/TabSelector/TabSelectorItem.js [hunk 1]
```diff
@@ -20,44 +20,58 @@ const propTypes = {
...
-                activeOpacity={activeOpacity}
-                inactiveOpacity={inactiveOpacity}
-            />
-            <TabLabel
-                title={title}
-                activeOpacity={activeOpacity}
-                inactiveOpacity={inactiveOpacity}
-            />
+            {({hovered}) => (
+                <View
```
[the remark this answers is not in the document]
Comment (wildan-m) [reply src/components/TabSelector/TabSelectorItem.js]: Sure. done.

**Comment by wildan-m** [comment]
@Ollyws Yes I did.

How do I test it? I didn't involve hoverable changes in this test, just made changes in the AnimatedPressableWithFeedback.

This is three attempts profiling result to move the tab from `Scan` to `Distance`

| PressableWithFeedback + Animated.View | AnimatedPressableWithFeedback |
|-|-|
|SCR-20230925-v6zSCR-20230925-v7dSCR-20230925-v7s|SCR-20230925-vbxSCR-20230925-vceSCR-20230925-vct|

**Comment by wildan-m** [comment]
@Ollyws here is the result after add hover

<details><summary>1. AnimatedPressableWithFeedback + Animated.View</summary>
SCR-20230929-b9h
SCR-20230929-b95
SCR-20230929-b8v
</details>

<details><summary>2. PressableWithFeedback + Animated.View</summary>
SCR-20230929-bet
SCR-20230929-bel
SCR-20230929-bdz
</details>

<details><summary>3. AnimatdPressableWithFeedback + View</summary>
SCR-20230929-bpf
SCR-20230929-boy
SCR-20230929-bom

</details>


# Summary
Option 1 consistently shows render time below 3ms
Option 2 can be better than option 1 but it can be worse > 3ms
Option 3 consistently renders below 2.5 ms

I think we need to restore back animated.view to view

**Comment by wildan-m** [comment]
@Ollyws this new option (4th) is really performant. it uses Hoverable and doesn't have the previous glitch.

```
        <View style={[styles.flex1]}>
            <Hoverable>{(hovered) => (
                <AnimatedPressableWithFeedback
                    accessibilityLabel={title}
                    style={[styles.tabSelectorButton,
                    { backgroundColor: Boolean(hoverBackgroundColor) && hovered && !isFocused ? hoverBackgroundColor : backgroundColor }
                    ]}
                    wrapperStyle={[styles.flex1]}
                    onPress={onPress}
                >
                    <TabIcon
                        icon={icon}
                        activeOpacity={hovered && !isFocused ? 1 : activeOpacity}
                        inactiveOpacity={hovered && !isFocused ? 0 : inactiveOpacity}
                    />
                    <TabLabel
                        title={title}
                        activeOpacity={hovered && !isFocused ? 1 : activeOpacity}
                        inactiveOpacity={hovered && !isFocused ? 0 : inactiveOpacity}
                    />
                </AnimatedPressableWithFeedback>
            )}</Hoverable>
        </View>
```

<summary><details>Test result</details>
SCR-20230929-qq7
SCR-20230929-qpt
SCR-20230929-qpg

</summary>

**Comment by wildan-m** [comment]
@Ollyws nope, [this one](https://github.com/Expensify/App/issues/26648#issuecomment-1731111414).

**Comment by wildan-m** [comment]
Sure, here is 4th option test video


https://github.com/Expensify/App/assets/11847279/8c4acda6-e64b-476b-b598-b0a93b839479

**Comment by wildan-m** [comment]
@Ollyws I couldn't find any other bug, did you find any? are you okay with this 4th option?

**Comment by wildan-m** [comment]
@Ollyws I can reproduce it when propose the previous solution [in this branch](https://github.com/wildan-m/App/tree/wildan/fix/26648-2-tab-hover) seems the actual issue is related to getBackground function call inside hoverable component

File: src/styles/styles.js [hunk 2]
```diff
@@ -3435,11 +3435,15 @@ const styles = (theme) => ({
 
     tabText: (isSelected) => ({
         marginLeft: 8,
-        fontFamily: isSelected ? fontFamily.EXP_NEUE_BOLD : fontFamily.EXP_NEUE,
```
[the remark this answers is not in the document]
Comment (wildan-m) [reply src/styles/styles.js:3438]: I have replied to it not long after your comment. Is it shown in your GH?
Screenshot 2023-10-03 at 18 10 23

[the remark this answers is not in the document]
Comment (wildan-m) [reply src/styles/styles.js:3438]: @Ollyws seems I experienced GH technical issue like everyone else [here](https://github.com/orgs/community/discussions/10369).

[the remark this answers is not in the document]
Comment (wildan-m) [reply src/styles/styles.js:3438]: @Ollyws, are you going to check on the design team in the previous slack thread or c+ internal group?

[the remark this answers is not in the document]
Comment (wildan-m) [reply src/styles/styles.js:3438]: @Ollyws any other feedback?

**Comment by wildan-m** [comment]
@Ollyws, what browser did you use? can't reproduce with chrome

**Comment by wildan-m** [comment]
@Ollyws just updated the code

**Comment by wildan-m** [comment]
@Ollyws at this point I agree that overriding animated component value can cause unexpected results like that firefox issue.

I think the best option is to pass hovered style to the tabicon and tablabel like what I proposed in [my proposal (alternative 2)](https://github.com/Expensify/App/compare/main...wildan-m:App:wildan/fix/26648-4-tab-hover)
since this is adopted from [this comment](https://github.com/Expensify/App/issues/26648#issuecomment-1738330090) and I added ios native fix, I have no issue splitting the bounty with the original author.

Another option is to get back to our 3rd option [here](https://github.com/Expensify/App/pull/28357#issuecomment-1740195859), but based on our test it will not as fast as using hoverable.

what is your opinion?

**Comment by wildan-m** [comment]
@Ollyws I can't reproduce it in Chrome and Safari, seems firefox only

File: src/styles/styles.js [hunk 3]
```diff
@@ -3445,11 +3445,19 @@ const styles = (theme) => ({
...
+        fontFamily: fontFamily.EXP_NEUE_BOLD,
+        fontWeight: fontWeightBold,
         color: isSelected ? theme.textLight : theme.textSupporting,
     }),
 
+    tabBackground: (hovered, isFocused, backgroundColor) => ({
+        backgroundColor: hovered && !isFocused ? theme.highlightBG : backgroundColor,
+    }),
+
+    tabActiveOpacity: (hovered, isFocused, activeOpacity) => (hovered && !isFocused ? {opacity: 1} : {opacity: activeOpacity}),
```
[the remark this answers is not in the document]
Comment (wildan-m) [reply src/styles/styles.js]: @Ollyws works for me too, just committed your suggestion, but I'm curious why this only occurred at Firefox

[the remark this answers is not in the document]
Comment (wildan-m) [reply src/styles/styles.js]: @Ollyws, unfortunately, I can still reproduce it but not as often as before, should we support Firefox?

[the remark this answers is not in the document]
Comment (wildan-m) [reply src/styles/styles.js]: I'm not sure that making this one function will make it easy to understand. If it should, I'm thinking of this function

function tabOpacity(hovered, isFocused, activeOpacity, inactiveOpacity, active) {
    const activeValue = active ? activeOpacity : inactiveOpacity;
    const inactiveValue = active ? inactiveOpacity : activeOpacity;

  return {
    opacity: hovered && !isFocused ? activeValue : inactiveValue
  };
}

what's your opinion?

[the remark this answers is not in the document]
Comment (wildan-m) [reply src/styles/styles.js]: @Ollyws it works, but I'm a little confused with the naming. I've updated the code, does that look good to you?

**Comment by wildan-m** [comment]
> Also could we revert [these changes](https://github.com/Expensify/App/pull/28357/commits/3b9681688f808b085239df146f73782568f81a67) if they're no longer necessary? Thanks.

Sure, reverted

**Comment by shawnborton** [comment]
That looks good, but cc'ing Danny (@Expensify/design) to confirm as well!

**Comment by dannymcclain** [comment]
Yup that looks right to me!

**Comment by wildan-m** [comment]
Thanks for the confirmation. @Ollyws any other feedback?

**Comment by wildan-m** [comment]
@Ollyws that's weird, isn't it?

Even when we pass the hovered param to tabicon/label, the issue is occurred ([Alternative 2](https://github.com/wildan-m/App/tree/wildan/fix/26648-4-tab-hover))

For me, the only working solution that resolves all of these issues is this one: (don't forget to import StyleSheet from RN)

```
       <AnimatedPressableWithFeedback
            accessibilityLabel={title}
            style={[styles.tabSelectorButton, { backgroundColor }]}
            wrapperStyle={[styles.flex1]}
            onPress={onPress}
        >
            {({ hovered }) => (
                <View
                    style={[styles.tabSelectorButton, StyleSheet.absoluteFill, styles.tabBackground(hovered, isFocused, backgroundColor)]}
                >
                    <TabIcon
                        icon={icon}
                        activeOpacity={styles.tabOpacity(hovered, isFocused, activeOpacity, inactiveOpacity)}
                        inactiveOpacity={styles.tabOpacity(hovered, isFocused, inactiveOpacity, activeOpacity)}
                    />
                    <TabLabel
                        title={title}
                        activeOpacity={styles.tabOpacity(hovered, isFocused, activeOpacity, inactiveOpacity)}
                        inactiveOpacity={styles.tabOpacity(hovered, isFocused, inactiveOpacity, activeOpacity)}
                    />
                </View>
            )}
        </AnimatedPressableWithFeedback>
```

**Comment by wildan-m** [comment]
@Ollyws updated

**Comment by wildan-m** [comment]
@Ollyws based on [this test](https://github.com/Expensify/App/pull/28357#issuecomment-1740195859). Our recent changes are slightly better in terms of performance. Your proposed changes are option 2. Are you sure to apply it?

**Comment by wildan-m** [comment]
@Ollyws sure, change has been applied

**Review by Ollyws — APPROVED** [review]
LGTM.
Thanks for all the changes!

**Review by MariaHCD — APPROVED** [review]

**Pull request merged**